src/components/CurrentCityWeather/CurrentCityWeather.js

```diff
@@ -100,19 +111,59 @@ const CurrentCityWeather = () => {
     setCity(addressComponents[0]);
   }
 
+
+  useEffect(() => {
+    if (transcript !== '') {
+      fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${transcript}&key=${process.env.REACT_APP_GOOGLE_MAPS_API_KEY}`)
+        .then(response => response.json())
+        .then(data => {
+          var latitude = data.results[0].geometry.location.lat;
+          var longitude = data.results[0].geometry.location.lng;
+          var coordinates = { lat: latitude, lon: longitude };
+          setAddress(data.results[0].formatted_address)
+          setCityCoordinates(coordinates);
+        })
+        .catch(error => alert(error))
+    }
+  }, [transcript])
+
+  function handleMicrophone() {
+    if (listening) {
+      SpeechRecognition.stopListening()
+    } else {
+      SpeechRecognition.startListening()
+    }
+  }
+
+  async function checkBraveBrowser() {
+    if (navigator.brave && await navigator.brave.isBrave()) {
+      return false
+    } else {
+      return true
+    }
```

Comment:
To check a boolean and return a boolean is a little bit redundant. Try this:

```suggestion
    return navigator.brave && await navigator.brave.isBrave();
```

Reply:
Ya this way is better I will change to this